### ISSUE TITLE
Add a table of contents

### ISSUE THREAD

**Issue by strboul** [ropensci/webservices#456]
Is it possible to add a table of contents (ToC) to the beginning of `README.md`? Displaying only the sections and subsections would be sufficient to have a clear overview in the document.

**Comment by sckott** [comment]
maybe, but they won't want that on the version that get's sent to CRAN - so it's creation and removal would have to be automated as part of Makefile workflow. do you want to try to make it?

**Comment by strboul** [comment]
Sure, let me work on it.

**Comment by strboul** [comment]
I've found a solution in the pandoc wiki page [here](https://github.com/jgm/pandoc/wiki/Pandoc-Tricks#toc-generation). To achieve this, first, I've replaced the pandoc line in the Makefile's `README.md` target with the one below:
`pandoc --toc -V toc-title:"Index" --template=readme-template.markdown --toc-depth=1 -o README.md WebTechnologies.html`

Second, created a pandoc template file named `readme-template.markdown` and fill it with the following content:
```
$if(toc)$
$if(toc-title)$
$toc-title$
$endif$
$toc$
$endif$
$body$
```
However, this does not show all headers properly as that conversion is known not to play quite nicely with the HTML tags such as `<div>`s. Some modification in the source HTML file seems required. And another problem is that ToC should appear after the main title and there is no clear way to do this with pandoc. I don't know but there may be some better ways to generate ToC directly from HTML.
